### DISCUSSION TITLE
Non able to read a JSON file: unexpected end of data. Try increasing "maximum_object_size"

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#6747] [Q&A] [answered]
Hi,
I have [this JSON file](https://github.com/duckdb/duckdb/files/10989110/tmp_b.json.gz).

If I run 

```sql
select * EXCLUDE (COMPONENTI_AIUTO) from  read_json('tmp_b.json.gz', json_format='array_of_records', auto_detect=true) limit 2;
```

I have "Error: Invalid Input Error: Malformed JSON in file "tmp_b.json.gz", at byte 2097149 in object 2: unexpected end of data. Try increasing "maximum_object_size"."

The JSON file should be valid and the command seems to me right. What am I doing wrong?

Thank you

**Comment by Mytherin** [comment, marked as answer]
This is likely related to #6569, the issue is that array_of_records data is currently parsed as one single JSON object (because effectively that is what it is). This can cause issues with larger JSON files stored in this format. You can increase the maximum allowed object size using the `maximum_object_size=X` parameter.

**Reply by aborruso** [reply]
Thank you very much. I have solved converting it in a jsonlines
